sdk/python/cmd/pulumi-language-python/main.go

```diff
@@ -704,7 +709,12 @@ func (host *pythonLanguageHost) Run(ctx context.Context, req *pulumirpc.RunReque
 			// If the program ran, but exited with a non-zero error code.  This will happen often, since user
 			// errors will trigger this.  So, the error message should look as nice as possible.
 			if status, stok := exiterr.Sys().(syscall.WaitStatus); stok {
-				err = errors.Errorf("Program exited with non-zero exit code: %d", status.ExitStatus())
+				switch status.ExitStatus() {
```

Comment:
I would love a `case 0: panic("unreachable")`

Reply:
I think I'll copy how nodejs handles it. thanks for pointing it out.

https://github.com/pulumi/pulumi/blob/0b86f575f20a51877e078ab7993aa65e088cd9cc/sdk/nodejs/cmd/pulumi-language-nodejs/main.go#L584-L587